deps/linenoise/linenoise.c

```diff
@@ -708,6 +708,31 @@ void linenoiseEditMoveRight(struct linenoiseState *l) {
     }
 }
 
+/* Consider letters/digits/underscore as “word”; others as delimiters. */
+static int isWordChar(char c) {
+    return (c == '_' || (c >= '0' && c <= '9') ||
+            (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'));
+}
+
+static void linenoiseEditMoveWordLeft(struct linenoiseState *l) {
+    if (l->pos == 0) return;
+    /* Move cursor to the left over any delimiters */
+    while (l->pos > 0 && !isWordChar(l->buf[l->pos - 1])) l->pos--;
+    /* Then continue moving over a word */
+    while (l->pos > 0 && isWordChar(l->buf[l->pos - 1])) l->pos--;
```

Comment:
```suggestion
    while (l->pos > 0 && l->isWordChar(l->buf[l->pos - 1])) l->pos--;
```
If we want to keep it generic, we might consider allowing linenoise to take in a callback for what a "word" is.

Reply:
Good call. However, instead of adding a new field `isWordChar` into the `linenoiseState` object, I end up with just passing callback functions. Lemme know if you like such design.

Reply:
Seems fine to me.